core/types/rlp_backwards_compat.libevm_test.go

```diff
@@ -106,3 +109,196 @@ func testHeaderRLPBackwardsCompatibility(t *testing.T) {
 		assert.Equal(t, hdr, got)
 	})
 }
+
+func TestBodyRLPBackwardsCompatibility(t *testing.T) {
+	rng := ethtest.NewPseudoRand(0)
+
+	randTx := func() *Transaction { return NewTx(&LegacyTx{Nonce: rng.Uint64()}) }
+	randHdr := func() *Header { return &Header{ParentHash: rng.Hash()} }
+	randWithdraw := func() *Withdrawal { return &Withdrawal{Index: rng.Uint64()} }
+
+	// We build up test-case [Body] instances from the power set of each of
+	// these components.
+	txs := [][]*Transaction{
+		nil, {}, // Must be equivalent for non-optional field
+		{randTx()},
+		{randTx(), randTx()}, // Demonstrates nested lists
+	}
+	uncles := [][]*Header{
+		nil, {},
+		{randHdr()},
+		{randHdr(), randHdr()},
+	}
+	withdrawals := [][]*Withdrawal{
+		nil, {}, // Must be different for optional field
+		{randWithdraw()},
+		{randWithdraw(), randWithdraw()},
+	}
+
+	var bodies []*Body
+	for _, t := range txs {
+		for _, u := range uncles {
+			for _, w := range withdrawals {
+				bodies = append(bodies, &Body{t, u, w})
+			}
+		}
+	}
+
+	for _, body := range bodies {
+		t.Run("", func(t *testing.T) {
+			t.Logf("%T: %+[1]v", body)
+
+			// The original [Body] doesn't implement [rlp.Encoder] nor
+			// [rlp.Decoder] so we can use a methodless equivalent as the gold
+			// standard.
+			type withoutMethods Body
+			wantRLP, err := rlp.EncodeToBytes((*withoutMethods)(body))
+			require.NoErrorf(t, err, "rlp.EncodeToBytes([%T with methods stripped])", body)
+
+			t.Run("Encode", func(t *testing.T) {
+				got, err := rlp.EncodeToBytes(body)
+				require.NoErrorf(t, err, "rlp.EncodeToBytes(%#v)", body)
+				assert.Equalf(t, wantRLP, got, "rlp.EncodeToBytes(%#v)", body)
+			})
+
+			t.Run("Decode", func(t *testing.T) {
+				got := new(Body)
+				err := rlp.DecodeBytes(wantRLP, got)
+				require.NoErrorf(t, err, "rlp.DecodeBytes(%v, %T)", wantRLP, got)
+
+				want := body
+				// Regular RLP decoding will never leave these non-optional
+				// fields nil.
+				if want.Transactions == nil {
+					want.Transactions = []*Transaction{}
+				}
+				if want.Uncles == nil {
+					want.Uncles = []*Header{}
+				}
+
+				opts := cmp.Options{
+					cmpeth.CompareHeadersByHash(),
+					cmpeth.CompareTransactionsByBinary(t),
+				}
+				if diff := cmp.Diff(body, got, opts); diff != "" {
+					t.Errorf("rlp.DecodeBytes(rlp.EncodeToBytes(%#v)) diff (-want +got):\n%s", body, diff)
+				}
+			})
+		})
+	}
+}
+
+// cChainBodyExtras carries the same additional fields as the ava-labs/coreth
+// [Body] and implements [BodyHooks] to achieve equivalent RLP {en,de}coding.
+type cChainBodyExtras struct {
```

Comment:
nit: why the name `cChainBodyExtras` by the way (as in, why the `c` prefix)?

Reply:
As in Avalanche C-Chain. I'll update the comment to clarify.